crates/ntx-builder/src/builder.rs

```diff
@@ -177,26 +179,46 @@ impl NetworkTransactionBuilder {
         event: MempoolEvent,
         account_actor_config: &AccountActorConfig,
         chain_state: Arc<RwLock<ChainState>>,
-    ) {
+    ) -> Result<(), anyhow::Error> {
         match &event {
-            MempoolEvent::TransactionAdded { account_delta, .. } => {
-                if let Some(AccountUpdateDetails::New(account)) = account_delta {
-                    // Spawn new actors if a transaction creates a new network account
-                    if let Some(network_account) = AccountOrigin::transaction(account) {
-                        self.coordinator.spawn_actor(network_account, account_actor_config);
-                    }
-                } else {
-                    self.coordinator.broadcast_event(&event).await;
+            MempoolEvent::TransactionAdded { id, account_delta, .. } => {
+                match account_delta {
+                    Some(AccountUpdateDetails::New(account)) => {
+                        // Create new actor for account creation transactions.
+                        if let Some(network_account) = AccountOrigin::transaction(account) {
+                            self.coordinator
+                                .spawn_actor(network_account, account_actor_config)
+                                .await?;
+                        }
+                        Ok(())
+                    },
+                    Some(AccountUpdateDetails::Delta(account_delta)) => {
+                        // Cache notes that predate corresponding accounts if there are any.
+                        if let Ok(prefix) = NetworkAccountPrefix::try_from(account_delta.id()) {
+                            self.coordinator.cache_predating_events(prefix, &event, id);
+                        }
```

Comment:
@bobbinth this is probably not sufficient. I suppose I need to search the prefixes involved within every note of the transaction? Not just the one pertaining to the account delta?

Reply:
Yes, in fact, I don't think the account from account delta is relevant in this context. Basically, for every transaction we need to get a list of output notes (we only care about network notes addressed to a single account), and then route them to their respective accounts. The notes that remain "unassigned" after this process should be treated separately (i.e., put into some pool from which they can be assigned to an account if/when it gets created).

Reply:
The model we have right now is that we broadcast every event to the actors. Initially we did instead filter out irrelevant actors but we removed that logic for some reason (lost in sea of resolved comments in first PR).

This [commit](https://github.com/0xMiden/miden-node/pull/1219/commits/5ec11456e4e036bdf2f3f1b308c78133dad64472) shows the logic we removed. Essentially:
```rust
    fn find_affected_accounts(
        account_delta: Option<&AccountUpdateDetails>,
        network_notes: &[NetworkNote],
    ) -> HashSet<NetworkAccountPrefix> {
```

We might want to go back to this both for this PR and in the future if we decide to only spawn ephemeral tasks rather than keeping a task around for every actor.